ansible/roles/preflights/tasks/main.yaml

```diff
@@ -0,0 +1,393 @@
+---
+  - name: check for supported OS
+    fail:
+      msg: "{{ configured_ansible_distribution }} {{ ansible_distribution_major_version }} is not supported. The supported OSes - CentOS 7/8, RHEL 7/8, Suse 15, SLES 15, Ubuntu 16 (xenial) 18 (bionic) or Debian 9 (stretch)"
+    failed_when: not (
+                 (configured_ansible_distribution == "CentOS" and ansible_distribution_major_version|int in [7, 8])
+                 or (configured_ansible_distribution == "RedHat" and ansible_distribution_major_version|int in [7, 8])
+                 or (configured_ansible_distribution == 'Ubuntu' and ((ansible_distribution_major_version|int == 16) or (ansible_distribution_major_version|int == 18)))
+                 or (configured_ansible_distribution == 'Debian' and ((ansible_distribution_major_version|int == 9) or (ansible_distribution_major_version|int == 10)))
+                 or (ansible_distribution == 'Suse' and ansible_distribution_major_version|int == 15)
+                 or (ansible_distribution == 'SLES' and ansible_distribution_major_version|int == 15)
+                 or (ansible_distribution == 'openSUSE Leap' and ansible_distribution_major_version|int == 15)
+                 )
+
+# TODO how to determine which ansible_mounts to use in docker?
+#  - name: fail when disk utilization more than 90%
+#    command: /bin/true
+#    changed_when: false
+#    failed_when: ((ansible_mounts[0].size_total - ansible_mounts[0].size_available)/ansible_mounts[0].size_total) > 0.9
+
+  - name: check required OS package registries are accessible for rpm
+    uri:
+      url: "{{item}}"
+      status_code: 200
+    with_items:
+      - "{{kubernetes_rpm_repository_url}}/repodata/repomd.xml"
+      - "{{kubernetes_rpm_gpg_key_url}}"
+      - "{{docker_rpm_container_selinux_package_url}}"
+      - "{{docker_rpm_container_selinux_gpg_key_url}}"
+      - "{{docker_rpm_repository_url}}/repodata/repomd.xml"
+      - "{{docker_rpm_gpg_key_url}}"
+      - "{{libnvidia_container_rpm_repository_package_url}}/repodata/repomd.xml"
+      - "{{libnvidia_container_gpg_key_url}}"
+      - "{{nvidia_container_runtime_rpm_repository_package_url}}/repodata/repomd.xml"
+      - "{{nvidia_container_runtime_gpg_key_url}}"
+    register: result
+    until: result is success
+    retries: 5
+    delay: 3
+    when:
+      - package_versions.enable_repository_installation
+      - ansible_os_family == 'RedHat'
+      - ansible_distribution_major_version|int < 8
+      - rpms_tar_file == ''
+
```

Comment:
When will these preflights run?
If it runs after `kib provision` has already run then this is no longer needed since the install process would be alraedy done.